pageserver/src/tenant.rs

```diff
@@ -1717,7 +1718,32 @@ impl Tenant {
         tenant_id: TenantId,
         remote_storage: Option<GenericRemoteStorage>,
     ) -> Tenant {
-        let (state, _) = watch::channel(state);
+        let (state, mut rx) = watch::channel(state);
+
+        tokio::spawn(async move {
+            let current_state = *rx.borrow_and_update();
+            TENANT_STATE_METRIC
+                .with_label_values(&[&tenant_id.to_string(), current_state.as_str()])
+                .inc();
+            loop {
+                match rx.changed().await {
+                    Ok(()) => {
+                        let new_state = *rx.borrow();
+                        TENANT_STATE_METRIC
+                            .with_label_values(&[&tenant_id.to_string(), current_state.as_str()])
+                            .dec();
+                        TENANT_STATE_METRIC
+                            .with_label_values(&[&tenant_id.to_string(), new_state.as_str()])
+                            .inc();
+                    }
+                    Err(_sender_dropped_error) => {
+                        info!("Tenant dropped the state updates sender, quitting waiting for tenant state change");
+                        return;
+                    }
+                }
+            }
```

Comment:
```suggestion
            let tid = tenant_id.to_string();
            TENANT_STATE_METRIC
                .with_label_values(&[&tid, current_state.as_str()])
                .inc();
            loop {
                match rx.changed().await {
                    Ok(()) => {
                        let new_state = *rx.borrow();
                        TENANT_STATE_METRIC
                            .with_label_values(&[&tid, current_state.as_str()])
                            .dec();
                        TENANT_STATE_METRIC
                            .with_label_values(&[&tid, new_state.as_str()])
                            .inc();
                    }
                    Err(_sender_dropped_error) => {
                        info!("Tenant dropped the state updates sender, quitting waiting for tenant state change");
                        return;
                    }
                }
            }
```

This will avoid unnecessary allocations.